infrastructure/db/handler.js

```diff
@@ -1,253 +1,208 @@
-                }
-            }
-            resolve(resp);
-        });
-    });
+/**
+* Finds Teammember by email
+* @param {string} emailAddress email of the wanted Teammember
+* @return {JSON} json, with Teammember's data
+*/
+const getTeammemberByEmail = async(emailAddress) => {
+    try {
+        return await models.teammember.findOne({ where : { emailAddress : emailAddress}});
+    } catch (err) {
+        throw Error(err);
+    }
 }
-
-exports.addTeamMemeber = async(request) => {
-    models.teammember.create({ emailAddress : request.emailAddress, 
-        password : request.password });
+/**
+* Creates Teammember
+* @param {JSON} member Details of the new members
+* @return Nothing.
+*/
+const createTeamMemeber = async(member) => {
+    models.teammember.create({ emailAddress : member.emailAddress, 
+        password : member.password });
 
 }
 
 /**
 * Finds all the Teammembers of specific board
 * @param {int} boardId id of the wanted board
-* @return {Promise} promise, with json response
+* @return {JSON} json, with board's teammembers details
 */
-exports.getBoardTeammembers = async(boardId) => {
-    
-    return new Promise((resolve) => {
-
-        models.board.findByPk(boardId).then((board) => {
-            board.getTeammembers().then(users => {
-
-                let resp = { Teammembers : new Array() };
-                for(let i = 0;i < users.length;i++) {
-                    resp.Teammembers[i] = {
-                        id : users[i].id,
-                        emailAddress : users[i].emailAddress,
-                        createdAt : users[i].createdAt,
-                        updatedAt : users[i].updatedAt
-                    }
-                }
-                resolve(resp);
-
-            });
-        });
-
-    });
+const getBoardTeammembers = async(boardId) => {
+    try {
+        const board = await getBoard(boardId);
+        if(board != null) {
+            return await board.getTeammembers();
+        } else {
+            return [];
+        }
+    } catch(err) {
+        throw Error(err);
+    }
 }
 /**
 * Finds all the Groups of specific board
 * @param {int} boardId id of the wanted board
-* @return {Promise} promise, with json response
+* @return {JSON} json with all groups of requested board
 */
-exports.getGroups = async(boardId) => {
-
-    return new Promise((resolve) => {
-
-        models.group.findAll({ where : { boardId : boardId } }).then(groups => {
-
-            let resp = { groups : new Array() }; //Builds array of all the groups
-            for(let i = 0;i < groups.length;i++) {
-                resp.groups[i] = {
-                    id : groups[i].id,
-                    label : groups[i].label,
-                    updatedAt : groups[i].updatedAt
-                }
-            }
-            resolve(resp);
-
-        });
-    });
+const getGroups = async(boardId) => {
+    try {
+        return await models.group.findAll({ where : { boardId : boardId } });
```

Comment:
Not important at all but with the standard of ES6 you can now use deconstruction which let's you define your where statement as:
`where: { boardId}` because the key `boardId` has the exact same naming for the associated value.